### ISSUE TITLE
LintCode url doesn't work

### ISSUE THREAD

**Issue by jonaslundquist** [invertase/dart_custom_lint#87]
**Describe the bug**
After upgrading to 0.2.x, it doesn't seem like `url` specified in LintCode isn't being reported for a lint. It doesn't show up for my custom lint and looking in the source code, as far as I can see, it's never used (seeing error.dart#120)

**To Reproduce**
create a lint with:
```
class MyCustomLintCode extends DartLintRule {
  MyCustomLintCode() : super(code: _code);

  /// Metadata about the warning that will show-up in the IDE.
  /// This is used for `// ignore: code` and enabling/disabling the lint
  static const _code = LintCode(
    name: 'my_custom_lint_code',
    problemMessage: 'This is the description of our custom lint',
    url: 'https://www.google.com'
  );

  @override
  void run(
    CustomLintResolver resolver,
    ErrorReporter reporter,
    CustomLintContext context,
  ) {
    // Our lint will highlight all variable declarations with our custom warning.
    context.registry.addVariableDeclaration((node) {
      // "node" exposes metadata about the variable declaration. We could
      // check "node" to show the lint only in some conditions.

      // This line tells custom_lint to render a waring at the location of "node".
      // And the warning shown will use our `code` variable defined above as description.
      reporter.reportErrorForNode(code, node);
    });
  }
}
```

**Expected behavior**
A `url` to go with the lint message. It used to work prior to 0.2. I'm using IntelliJ IDEA 2022.2.2.
Actual behavior: only `name` and `problemMessage` shows up.

Let me know if I can provide any additional info!
